src/tiktok_uploader/upload.py

```
@@ -522,28 +522,17 @@ def _remove_cookies_window(driver) -> None:
     """
 
     logger.debug(green("Removing cookies window"))
-    cookies_banner = WebDriverWait(driver, config.implicit_wait).until(
+    WebDriverWait(driver, config.implicit_wait).until(
         EC.presence_of_element_located(
             (By.TAG_NAME, config.selectors.upload.cookies_banner.banner)
         )
     )
 
-    item = WebDriverWait(driver, config.implicit_wait).until(
-        EC.visibility_of(
-            cookies_banner.shadow_root.find_element(
-                By.CSS_SELECTOR,
-                config.selectors.upload.cookies_banner.button,
-            )
-        )
-    )
-
-    # Wait that the Decline all button is clickable
-    decline_button = WebDriverWait(driver, config.implicit_wait).until(
-        EC.element_to_be_clickable(item.find_elements(By.TAG_NAME, "button")[0])
+    driver.execute_script(
+        "document.querySelector(arguments[0]).remove()",
```

Isn't there a way to do this with selenium and not using javascript apis?

---

> Isn't there a way to do this with selenium and not using javascript apis?

from selenium.webdriver.remote.webdriver import WebDriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

from tiktok_uploader import config, logger
from tiktok_uploader.utils import green, red

def _remove_cookies_window(driver: WebDriver) -> None:
    """
    Removes the cookies window if it is open by accessing the shadow DOM.

    Parameters
    ----------
    driver : selenium.webdriver
    """
    logger.debug(green("Removing cookies window"))
    try:
        # Wait for the shadow host (the banner) to be present
        shadow_host = WebDriverWait(driver, config.implicit_wait).until(
            EC.presence_of_element_located(
                (By.TAG_NAME, config.selectors.upload.cookies_banner.banner)
            )
        )

        # Access the shadow root
        shadow_root = shadow_host.shadow_root

        # Find the button wrapper inside the shadow root
        button_wrapper = shadow_root.find_element(
            By.CSS_SELECTOR, config.selectors.upload.cookies_banner.button
        )

        # Find the actual button within the wrapper (usually the first one is decline/reject)
        decline_button = button_wrapper.find_element(By.TAG_NAME, "button")

        # Wait for the button to be clickable and then click it
        WebDriverWait(driver, config.implicit_wait).until(
            EC.element_to_be_clickable(decline_button)
        )
        decline_button.click()

        logger.debug(green("Clicked the decline button in the cookies banner."))

    except TimeoutException:
        logger.debug(red("Cookies banner not found or button not clickable in time."))
    except Exception as e:
        logger.error(red(f"An unexpected error occurred while removing cookies window: {e}"))

---

I mean this is practically the same code as before, i decided to not use shadow root because it crash sometimes or maybe we can keep this logic and when it crash we try to remove it using the js script ?

---

> I mean this is practically the same code as before, i decided to not use shadow root because it crash sometimes or maybe we can keep this logic and when it crash we try to remove it using the js script ?

Normally, there's no problem with it. The shadow root is the recommended way now with Selenium 4. Let's try using the shadow root first, and then as a fallback, we can remove it if it fails. But tbh, using a JS script isn't recommended anymore, shadow root is the right way to go.

---

In my case, remove_cookie window using shadow root sometimes crash (for unknown reason) and the reason is shadow root is undefined something like that, so we can keep the shadow root logic and if it raise an error like not found we use the js script as a fallback

---

> In my case, remove_cookie window using shadow root sometimes crash (for unknown reason) and the reason is shadow root is undefined something like that, so we can keep the shadow root logic and if it raise an error like not found we use the js script as a fallback

Sounds good, let's do it that way. I'll test it on my side and I'll remove the js code once I see it's working for sure, since using js isn't recommended like I said before, at least for the sake of clean code.